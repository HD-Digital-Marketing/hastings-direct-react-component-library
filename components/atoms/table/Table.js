import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import './table.scss'

const Table = props => {

    const {
        tableHeaders,
        tableData,
        theme = '',
        className = ''
    } = props

    const tableClass = classNames('a-table', className, {
        'a-table--primary': theme === 'primary',
        'a-table--secondary': theme === 'secondary'
    })

    const headers = tableHeaders.map((value, index) =>
        <th key={index}>{value}</th>
    )

    const bodyItems = tableData.map((value, index) =>
        <tr key={index}>
            {value.map((value, index) =>
                <td key={index}>{value}</td>
            )}
        </tr>
    )

    return (
        <table className={tableClass}>
            <thead>
            <tr>
                {headers}
            </tr>
            </thead>
            <tbody>
                {bodyItems}
            </tbody>
        </table>
    );
}

Table.propTypes = {
    tableHeaders: PropTypes.array,
    tableData: PropTypes.array,
    theme: PropTypes.string,
    className: PropTypes.string,
}

export default Table;
