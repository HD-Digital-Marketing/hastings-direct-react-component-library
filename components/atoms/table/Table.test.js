import React from 'react';
import renderer from 'react-test-renderer';
import Table from './Table';

describe('Check the table renders', () => {
	test('it should match the snapshot', () => {

		const tableHeaders = ['First name', 'Last name', 'Age', 'Gender']

		const tableData = [['Jack', 'Smith', '29', 'Male'], ['Tom', 'Jones', '56', 'Male']]

		const tree = renderer
			.create(<Table tableHeaders={tableHeaders} tableData={tableData} theme="primary" />)
			.toJSON();
		expect(tree).toMatchSnapshot();

	});
});