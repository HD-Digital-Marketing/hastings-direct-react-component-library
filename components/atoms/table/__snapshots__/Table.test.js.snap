// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the table renders it should match the snapshot 1`] = `
<table
  className="a-table a-table--primary"
>
  <thead>
    <tr>
      <th>
        First name
      </th>
      <th>
        Last name
      </th>
      <th>
        Age
      </th>
      <th>
        Gender
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Jack
      </td>
      <td>
        Smith
      </td>
      <td>
        29
      </td>
      <td>
        Male
      </td>
    </tr>
    <tr>
      <td>
        Tom
      </td>
      <td>
        Jones
      </td>
      <td>
        56
      </td>
      <td>
        Male
      </td>
    </tr>
  </tbody>
</table>
`;
